src/material/core/option/option.html

```diff
@@ -1,19 +1,18 @@
-<mat-pseudo-checkbox *ngIf="multiple" class="mat-mdc-option-pseudo-checkbox"
-    [state]="selected ? 'checked' : 'unchecked'" [disabled]="disabled"></mat-pseudo-checkbox>
+<mat-pseudo-checkbox *ngIf="multiple" class="mat-mdc-option-pseudo-checkbox" [disabled]="disabled"
+    [state]="selected ? 'checked' : 'unchecked'" [attr.aria-hidden]="'true'"></mat-pseudo-checkbox>
```

Comment:
Does the `aria-hidden` do anything here? The element is basically just a `div`. Same for the other elements it was added to.

Reply:
Yes, the `aria-hidden` actually does something here :). When I tested, it helped make VoiceOver's behavior more consistent.

Without the aria-hidden, the pseudo checkbox here will produce an "empty" node in the accessibility tree. In chrome, it produces a "generic" node in the a11y tree that doesn't have any label or meaningful information.

Here is what the a11y tree looks like right now:

listbox "Toppings"
_option "Extra cheese"
__generic
__StaticText "Extra cheese"
__generic

While maybe screen readers *should* be able to ignore these extra generic nodes, but I have seen VoiceOver be confused by it. I've seen VoiceOver actually focus on the *checkbox* instead of the *option*. I was also seeing VoiceOver sometimes focusing on the *StaticText* instead of the *option* node. I think that is part of what was causing VoiceOver to read "Extra cheese, clikable" instead of the normal option with the selected state.

Adding `aria-hidden="true"` is the safe option in this case.

In general, I recommend we put `aria-hidden="true"` on our decorative elements. We have this problem in other components, and I'm trying to fix them as I come across them.

Reply:
I think we're good here, so I'll mark this as resolved. Please lmk if there are further questions.